src/SASjs.ts

```diff
@@ -577,19 +595,71 @@ export default class SASjs {
           extraResponseAttributes
         )
       }
-    } else if (
-      config.serverType === ServerType.Sas9 &&
-      config.username &&
-      config.password
-    ) {
-      return await this.sas9JobExecutor!.execute(sasJob, data, config)
     } else {
-      return await this.webJobExecutor!.execute(
-        sasJob,
-        data,
-        config,
-        loginRequiredCallback
-      )
+      return Promise.reject(new ErrorResponse(validationResult.msg))
+    }
+  }
+
+  /**
+   * this function validates the structure of input data and verify that proper rules are being followed for table name
+   *
+   * @param data a json object contains one or more table, it can also be null
+   * @returns a object which contains two attributes: 1) status: boolean, 2) msg: string
+   */
+  private validateInput(data: { [key: string]: any }): {
+    status: boolean
+    msg: string
+  } {
+    if (data === null) return { status: true, msg: '' }
+    for (const key in data) {
+      if (key.match(/^[a-zA-Z_]/)) {
+        if (key.match(/^[a-zA-Z_][a-zA-Z0-9_]*$/)) {
+          if (key.length <= 32) {
+            if (this.getType(data[key]) === 'Array') {
```

Comment:
My personal preference would be to escape the nesting for the sake of better readability.
My suggestion is to do something like this:

```
// On every false condition you return with an error

if (!key.match(/^[a-zA-Z_]/)) {
  return {
    status: false,
    msg: 'First letter of table should be alphabet or underscore.'
  }
}

if (!key.match(/^[a-zA-Z_][a-zA-Z0-9_]*$/)) {
  return { status: false, msg: 'Table name should be alphanumeric.' }
}

// Finally if all are true, none of them were called, so you are left with returning the results.

return { status: true, msg: '' }
```

Reply:
refactored.